### ISSUE TITLE
Wire service should use class property default values

### ISSUE THREAD

**Issue by Gr8Gatsby** [salesforce/lwc#264]
_SOMA#976_

### Migrated issue

notes:

### Description

Due to ES6 class lifecycle, the @wire setter hooks aren't in place when default property values are applied.

#### Steps to Reproduce

Consider this code:
```
export default class Component extends Element {
    constructor() {
        super();
        this.size = 0;
    }

    @wire('listUi', {listViewId:'00BR0000000tMwYMAU', pageSize:'$size'})
    listUi;

    nextPage() {
        this.size += 10;
    }
}
```

`this.size` is set only in the constructor which is before the @wire has hooked the setter. As a result it doesn't see the value.

NOTE that we don't want to use the default value if an @api overrides the value.

See TODO at https://git.soma.salesforce.com/lwc/lwc/blob/master/packages/lwc-wire-service/src/wire-service.js#L219

**Comment by caridy** [comment]
this will be solved by https://github.com/salesforce/lwc/pull/1417

**Comment by kevinv11n** [comment]
Fixed by #767
